teleport/datadog_checks/teleport/check.py

```diff
@@ -0,0 +1,44 @@
+# (C) Datadog, Inc. 2024-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+from datadog_checks.base import OpenMetricsBaseCheckV2
+
+COMMON_METRICS_MAP = {
+    'process_state': 'process.state',
+    'certificate_mismatch': 'certificate_mismatch',
+    'rx': 'rx',
+    'server_interactive_sessions_total': 'server_interactive_sessions_total',
+    'teleport_build_info': 'teleport.build_info',
+    'teleport_cache_events': 'teleport.cache_events',
+    'teleport_cache_stale_events': 'teleport.cache_stale_events',
+    'tx': 'tx',
+}
+
+METRIC_MAP = {
+    **COMMON_METRICS_MAP
+}
+class TeleportCheck(OpenMetricsBaseCheckV2):
+    __NAMESPACE__ = 'teleport'
+
+    def __init__(self, name, init_config, instances):
+        super().__init__(name, init_config, instances)
+        self.check_initializations.appendleft(self._parse_config)
+
+    def check(self, _):
+        try:
+            super().check(_)
+            response = self.http.get(self.diagnostic_url + "/healthz")
+            response.raise_for_status()
+            self.service_check("health.up", self.OK)
+        except Exception as e:
+            self.service_check("health.up", self.CRITICAL, message=str(e))
```

Comment:
Can we separate out the health check, so that if the check raises an exception before we can hit the health endpoint, we don't say the health is critical (we can say it's unknown in that case).